doc/style-guide-git.md

```diff
@@ -0,0 +1,21 @@
+Style Guide for Git
+===================
+
+This extends the [general programming](style-guide-programming.md) portion of the [Style Guide](style-guide.md).
+
+
+* Branches
```

Comment:
Many times I've had problems with people not cleaning up old branches.  Can we have a note about that or add to the last note as part of "maintain the branch"